### DISCUSSION TITLE
Iteration is reversed when nesting data

### DISCUSSION THREAD

**Discussion by agustavo87** [alpinejs/alpine#3469] [7. Bugs (archived)]
[JSFiddle](https://jsfiddle.net/px6rLk48/15/)
Related to [#2210](https://github.com/alpinejs/alpine/issues/2210)

**Comment by ekwoka** [comment]
Use keys and put a div between the templates.

**Reply by agustavo87** [reply]
In fact, it does have a key (`x-key="item.id + '_' + subitem.id"`) but that does not make a difference in this case.
Although using an itermediate element in this case could be simple. In some other cases like when using tables could be problematic, as is (almost) not possible to use a an element without breaking the structure. As in [this example with tables](https://jsfiddle.net/8me7ghfj/7/). That was my original case, and although it has a [solution](https://jsfiddle.net/8me7ghfj/9/), is not as expected by documentation, or deducible by it.

**Reply by ekwoka** [reply]
> In some other cases like when using tables could be problematic, as is (almost) not possible to use a an element without breaking the structure

You can use `display: contents` to make the wrapper a passthrough for the flow. Provided it is not a banned element, the browser will handle that fine.
